user/tests/wiring/no_fixture/fastpin.cpp

```diff
@@ -0,0 +1,40 @@
+
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+
+
+test(FASTPIN_01_MaxDuration_SetReset) {
+    // Attempt pinSetFast and pinResetFast numerous times and check speed.
+    // Duration went from 62 -> 24 ticks per set on Gen2 with this change list
+
+#if PLATFORM_ID == PLATFORM_ARGON || \
+    PLATFORM_ID == PLATFORM_BORON || \
+    PLATFORM_ID == PLATFORM_ASOM || \
+    PLATFORM_ID == PLATFORM_BSOM || \
+    PLATFORM_ID == PLATFORM_B5SOM || \
+    PLATFORM_ID == PLATFORM_TRACKER
+    // expected max ticks of pinSetFast / pinResetFast on Gen3
+    const uint32_t MAX_DURATION_PINSET_TICKS = 60;
+#else
+    // expected max ticks of pinSetFast / pinResetFast on Gen2
+    const uint32_t MAX_DURATION_PINSET_TICKS = 24;
+#endif
+
+    const uint32_t NUM_ITERATIONS = 100;
+    uint32_t start, finish;
+
+    ATOMIC_BLOCK()  {
+        start = System.ticks();
+        for (uint32_t i = 0; i < NUM_ITERATIONS; i++) {
+            pinSetFast(D7);
```

Comment:
Nit: I would time these separately. I saw some variation in timings between them, might be good idea to watch out for both separately.